File: library/src/main/java/com/getstream/sdk/chat/rest/request/BanUserRequest.java
package com.getstream.sdk.chat.rest.request;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class BanUserRequest {

    @NotNull
    @Expose
    @SerializedName("target_user_id")
    String targetUserId;

    @Nullable
    @Expose
    @SerializedName("timeout")
    Integer timeout;

    @Nullable
    @Expose
    @SerializedName("reason")
    String reason;

    @Nullable
    @Expose
    @SerializedName("type")
    String channelType;

    @Nullable
    @Expose
    @SerializedName("id")
    String channelId;

    public BanUserRequest(@NotNull String targetUserId, @Nullable Integer timeout,
                          @Nullable String reason, @Nullable String channelType, @Nullable String channelId) {
        this.targetUserId = targetUserId;
        this.timeout = timeout;
        this.reason = reason;
        this.channelType = channelType;
        this.channelId = channelId;
    }
}

File: library/src/main/java/io/getstream/chat/android/client/rest/BanUserRequest.kt
package io.getstream.chat.android.client.rest

import com.google.gson.annotations.SerializedName


data class BanUserRequest(
    @SerializedName("target_user_id")
    var targetUserId: String,
    var timeout: Int,
    var reason: String,
    @SerializedName("type")
    var channelType: String,
    @SerializedName("id")
    var channelId: String
)